### PR TITLE
OPSEXP-1803 Add Search Enteprise reindex service

### PR DESCRIPTION
**Pull request by gionn**
OPSEXP-1803

### CODE REVIEW

File: .ansible-lint [hunk 1]
```diff
@@ -1,6 +1,5 @@
 warn_list:
   - experimental
-  - deprecations
```
Comment (alxgomz) [inline .ansible-lint:3]: 👍

File: roles/search_enterprise/templates/elasticsearch-cfg.j2 [hunk 2]
```diff
@@ -0,0 +1,9 @@
+Environment=ALFRESCO_SHAREDFILESTORE_BASEURL=http://{{ sfs_host }}:{{ ports_cfg.sfs.http }}/alfresco/api/-default-/private/sfs/versions/1/file
+Environment=ALFRESCO_ACCEPTED_CONTENT_MEDIA_TYPES_CACHE_BASE_URL=http://{{ trouter_host }}:{{ ports_cfg.transformers.trouter }}/transform/config
+Environment=ELASTICSEARCH_INDEXNAME=alfresco
+Environment=SPRING_ACTIVEMQ_BROKERURL=failover:({{ activemq_protocol }}://{{ activemq_host }}:{{ ports_cfg.activemq.openwire }})
```
Comment (alxgomz) [inline roles/search_enterprise/templates/elasticsearch-cfg.j2:4]:
```suggestion
Environment=SPRING_ACTIVEMQ_BROKERURL=failover:({{ activemq_protocol }}://{{ activemq_host }}:{{ ports_cfg.activemq[activemq_protocol | default('openwire')] }})
```

File: molecule/elasticsearch/verify.yml [hunk 3]
```diff
@@ -0,0 +1,54 @@
+- name: Verify
+  hosts: all
```
Comment (alxgomz) [inline molecule/elasticsearch/verify.yml:2]:
```suggestion
  hosts: search_enterprise
```

File: molecule/elasticsearch/verify.yml [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    - name: "Check that {{ job_name }} is really up and running"
+      become: true
+      ansible.builtin.shell: journalctl -u {{ job_name }}
+      register: error_log
+      retries: 6
+      delay: 10
+      until: error_log.stdout | regex_search('Started LiveIndexingApp')
+
+- name: Verify Reindex before first run
+  hosts: all
```
Comment (alxgomz) [inline molecule/elasticsearch/verify.yml:25]:
```suggestion
  hosts: search_enterprise
```

File: molecule/elasticsearch/verify.yml [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+      ansible.builtin.assert:
+        that:
+          - ansible_facts.services[job_name] is defined
+          - ansible_facts.services[job_name].state == 'inactive'
+
+- name: Run the Elasticsearch Connector Reindex
+  import_playbook: ../../playbooks/search-enterprise-reindex.yml
+
+- name: Verify Reindex after first run
+  hosts: all
```
Comment (alxgomz) [inline molecule/elasticsearch/verify.yml:40]:
```suggestion
  hosts: search_enterprise[0]
```

File: docs/deployment-guide.md [hunk 6]
```diff
@@ -833,6 +834,18 @@ In that case, you can add the `cluster_keepoff` variable to one of the `reposito
...
+## Maintenance
+
+After the initial deploy, may arise the need to execute maintenance tasks that are handled via specific playbooks.
+
+### Search Enterprise Reindexing
+
+You can trigger the reindexing of existing content in Search Enterprise using a dedicated playbook:
+
+```bash
+pipenv run ansible-playbook playbooks/search-enterprise-reindex.yml -i <inventory_file>.yml
```
Comment (alxgomz) [inline docs/deployment-guide.md:846]: We should add a warning message that if the reindex is triggerred on an index that has already indexed content that may/will create duplicates...
Wondering if we should also add a safe guard in the service file so it's not fired up manually by an admin who may actually want to start the connector itself?

**Review by alxgomz — CHANGES_REQUESTED** [review]

File: docs/deployment-guide.md (hunk 6 above)
Comment (gionn) [reply docs/deployment-guide.md:846]: I think that this type of concerns should be documented somewhere like https://docs.alfresco.com/search-enterprise/latest/admin/ and not here

can't find anything related to duplication search in the docs

Comment (gionn) [reply docs/deployment-guide.md:846]: anyway I added support for specifying additional parameters for the reindex job so it's possibile to do partial reindexing as per docs

File: roles/repository/templates/alfresco-global.properties.j2 [hunk 7]
```diff
@@ -70,7 +70,7 @@ tika.url=${localTransform.core-aio.url}
 transform.misc.url=${localTransform.core-aio.url}
 sfs.url=http://{{ sfs_host }}:{{ ports_cfg.sfs.http }}/
 
-messaging.broker.url=failover:({{ activemq_protocol }}://{{ activemq_host }}:{{ ports_cfg.activemq.openwire }})?timeout=3000
+messaging.broker.url=failover:({{ activemq_transport }}://{{ activemq_host }}:{{ ports_cfg.activemq[activemq_protocol] }})?timeout=3000
```
Comment (alxgomz) [inline roles/repository/templates/alfresco-global.properties.j2:73]: 👍

File: docs/deployment-guide.md (hunk 6 above)
Comment (alxgomz) [reply docs/deployment-guide.md:846]: I'm still concerned about someone mistakenly triggering the reindex through a `systemctl` command.
We could just check for existence of a file or var before starting that process (ofc the playbook would need to set things up when invoked on purpose).

Comment (gionn) [reply docs/deployment-guide.md:846]: engineers reported that running multiple times the reindex is not a problem, so we should be good to go as it is

**Review by alxgomz — APPROVED** [review]

**Pull request merged**